daemons/compute/client-go/main.go

```diff
@@ -80,26 +83,46 @@ func main() {
 		if err != nil {
 			log.Fatalf("could not create data movement request: %v", err)
 		}
+		uid := createResponse.GetUid()
 
 		if createResponse.GetStatus() == pb.DataMovementCreateResponse_SUCCESS {
 			log.Printf("Data movement request created: %s", createResponse.GetUid())
 		} else {
 			log.Fatal("Create request failed: ", createResponse.String())
 		}
 
-		// Wait for request to be completed
+		// Wait for request to be completed/cancelled
+		count := 0
+		cancelled := false
 		for {
-			statusResponse, err := getStatus(ctx, c, *workflow, *namespace, createResponse.GetUid(), *maxWaitTime)
+			statusResponse, err := getStatus(ctx, c, *workflow, *namespace, uid, *maxWaitTime)
 			if statusResponse.GetStatus() == pb.DataMovementStatusResponse_FAILED {
 				log.Fatalf("Data movement failed: %v", err)
 			}
 
 			log.Printf("Data movement status %s", statusResponse.String())
-			if statusResponse.GetState() == pb.DataMovementStatusResponse_COMPLETED {
+			if cancelled {
+				if statusResponse.GetState() == pb.DataMovementStatusResponse_COMPLETED &&
+					statusResponse.GetStatus() == pb.DataMovementStatusResponse_CANCELLED {
+					break
+				}
+			} else if statusResponse.GetState() == pb.DataMovementStatusResponse_COMPLETED {
 				break
 			}
 
+			// Cancel the data movement after some iterations
+			if *cancelRequests && !cancelled && count >= CancelAfterDurationInSeconds {
+				log.Printf("Canceling request: %v", uid)
+				cancelResponse, err := cancelRequest(ctx, c, *workflow, *namespace, uid)
+				if err != nil {
+					log.Fatalf("error initiating data movement cancel request: %v", err)
+				}
+				log.Printf("Data movement request cancel initiated: %v %v", uid, cancelResponse.String())
+				cancelled = true
```

Comment:
I don't think you need this flag - you can just check the status.Response.State == Completed and always break.